### PR TITLE
Bump operator-sdk to 1.17

### PR DESCRIPTION
**Pull request by pavolloffay**
This PR bumps operator-skd from 1.13 to 1.17. It follows the SDK upgrade docs and aligns project with those changes.

Notable changes:
* tools are installed to `./bin`
* unit tests that use `envtest` require additional settings based on the k8s version used in the makefile - added contributing docs
* Removed support from `admissionregistration.k8s.io/v1beta1`, now only `v1` is supported (as part of operator-sdk 1.14 upgrade). The  `v1beta1` was removed in 1.22 and deprecated in 1.16 (added `v1`). OpenShift 4.6 LTS is based on k8s 1.19.

1.17 update
*  https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/#gov3-upgrade-go-and-dependencies

1.14 update
* https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.14.0/#upgrade-k8s-versions-to-use-122-golangv3

Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

### CODE REVIEW

File: config/default/manager_auth_proxy_patch.yaml [hunk 1]
```diff
@@ -15,11 +15,18 @@ spec:
         - "--secure-listen-address=0.0.0.0:8443"
         - "--upstream=http://127.0.0.1:8080/"
         - "--logtostderr=true"
-        - "--v=10"
+        - "--v=0"
```
Comment (pavolloffay) [inline config/default/manager_auth_proxy_patch.yaml:18]: https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/#reduce-debug-log-level-for-the-sidecar-container-kube-rbac-proxy-from-10-to-0

File: bundle/manifests/opentelemetry.io_instrumentations.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.6.0-beta.0
+    controller-gen.kubebuilder.io/version: v0.8.0
```
Comment (pavolloffay) [inline bundle/manifests/opentelemetry.io_instrumentations.yaml:5]: https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/#gov3-upgrade-go-and-dependencies

File: apis/v1alpha1/instrumentation_webhook.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ func (r *Instrumentation) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
-//+kubebuilder:webhook:path=/mutate-opentelemetry-io-v1alpha1-instrumentation,mutating=true,failurePolicy=fail,sideEffects=None,groups=opentelemetry.io,resources=instrumentations,verbs=create;update,versions=v1alpha1,name=minstrumentation.kb.io,admissionReviewVersions={v1,v1beta1}
+//+kubebuilder:webhook:path=/mutate-opentelemetry-io-v1alpha1-instrumentation,mutating=true,failurePolicy=fail,sideEffects=None,groups=opentelemetry.io,resources=instrumentations,verbs=create;update,versions=v1alpha1,name=minstrumentation.kb.io,admissionReviewVersions=v1
```
Comment (pavolloffay) [inline apis/v1alpha1/instrumentation_webhook.go:45]: https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.14.0/#upgrade-k8s-versions-to-use-122-golangv3

**Comment by pavolloffay** [comment]
@VineethReddy02 @jpkrohling this is ready for review.

**Comment by jpkrohling** [comment]
Can you ping me again once the CI passes? I'll then review and merge this PR.

**Comment by pavolloffay** [comment]
@jpkrohling all green now!

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -408,46 +412,45 @@ spec:
...
-    webhookPath: /mutate-opentelemetry-io-v1alpha1-opentelemetrycollector
+    webhookPath: /mutate-v1-pod
   - admissionReviewVersions:
     - v1
-    - v1beta1
     containerPort: 443
     deploymentName: opentelemetry-operator-controller-manager
-    failurePolicy: Ignore
-    generateName: vinstrumentationdelete.kb.io
+    failurePolicy: Fail
```
Comment (jpkrohling) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:444]: I see that this is not a change in behavior, but we might want to reconsider in a follow up issue/PR to change this to `Ignore`. As it is, we'd block people from creating pods if the operator has a bug or isn't available.

File: pkg/collector/upgrade/upgrade_test.go [hunk 5]
```diff
@@ -48,10 +48,10 @@ func TestShouldUpgradeAllToLatestBasedOnUpgradeStrategy(t *testing.T) {
 			// prepare
 			nsn := types.NamespacedName{Name: "my-instance", Namespace: "default"}
 			existing := makeOtelcol(nsn)
-			existing.Status.Version = beginV
 			err := k8sClient.Create(context.Background(), &existing)
 			require.NoError(t, err)
 
+			existing.Status.Version = beginV
```
Comment (jpkrohling) [inline pkg/collector/upgrade/upgrade_test.go:54]: What's the story behind this change?

Comment (pavolloffay) [reply pkg/collector/upgrade/upgrade_test.go:54]: The story is that after upgrading the dependencies the test was not working. Apparently the create call was reverting the version to empty string.

Comment (jpkrohling) [reply pkg/collector/upgrade/upgrade_test.go:54]: Alright, so, a bug in the test?

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml (hunk 4 above)
Comment (pavolloffay) [reply bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:444]: Yes this is not related to this PR, the webhook for the collector use fail policy https://github.com/open-telemetry/opentelemetry-operator/blob/abe7d5358997f9cb0a16f6115b652fc47e51b3c6/apis/v1alpha1/opentelemetrycollector_webhook.go#L37.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -88,6 +88,17 @@ USE_EXISTING_CLUSTER=true make test
 
 Tests can also be run without an existing cluster. For that, install [`kubebuilder`](https://book.kubebuilder.io/quick-start.html#installation). In this case, the tests will bootstrap `etcd` and `kubernetes-api-server` for the tests. Run against an existing cluster whenever possible, though.
 
+### Unit tests
+
+Some unit tests use [envtest](https://book.kubebuilder.io/reference/envtest.html) which requires Kubernetes binaries (e.g. `api-server`, `etcd` and `kubectl`) to be present on the host filesystem. Makefile takes care of installing all dependent binaries, however running the tests from IDE or via `go test` might not work out-of-the-box. The `envtest` uses env variable `KUBEBUILDER_ASSETS` that points to a directory with these binaries. To make the test work in IDE or `go test` the environment variable has to be correctly set.
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply CONTRIBUTING.md:93]: Interesting, this might be something new then.

**Review by jpkrohling — APPROVED** [review]

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml (hunk 4 above)
Comment (jpkrohling) [reply bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:444]: Might still be worth reconsidering in a follow-up PR/issue.

**Pull request merged**